vcd/resource_vcd_rde.go

```diff
@@ -0,0 +1,435 @@
+// (the resource must be already defined in .tf config otherwise `terraform import` will complain)
+// 5. `terraform refresh` is being implicitly launched. The Read method looks up all other fields
+// based on the known ID of object.
+//
+// Example resource name (_resource_name_): vcd_rde.outer-rde
+// Example import path (_the_id_string_): my-rde.vmware.kubernetes.1.0.0
+// Note: the separator can be changed using Provider.import_separator or variable VCD_IMPORT_SEPARATOR
+func resourceVcdRdeImport(_ context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	vcdClient := meta.(*VCDClient)
+	helpError := fmt.Errorf(`resource id must be specified in one of these formats:
+'rde-id' to import by RDE id
+'vendor.nss.version.name.position' where position is the RDE number as returned by VCD, starting on 1
+'list@vendor.nss.version.name' to get a list of RDEs with their respective positions and real IDs`)
+
+	printList := func(vendor, nss, version, name string) error {
+		rdes, err := vcdClient.GetRdesByName(vendor, nss, version, name)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Found RDEs with vendor '%s', nss '%s', version '%s' and name '%s':\n", vendor, nss, version, name)
+		for _, rde := range rdes {
+			fmt.Printf("* %s\n", rde.DefinedEntity.ID)
+		}
+		return fmt.Errorf("resource was not imported! %s", helpError.Error())
+	}
+
+	getRdeInPosition := func(vendor, nss, version, name, position string) (*govcd.DefinedEntity, error) {
+		rdes, err := vcdClient.VCDClient.GetRdesByName(vendor, nss, version, name)
+		if err != nil {
+			return nil, err
+		}
+		idx, err := strconv.Atoi(position)
+		if err != nil {
+			return nil, fmt.Errorf("introduced position %v is not a number", position)
+		}
+		if idx < 1 || idx > len(rdes) {
+			return nil, fmt.Errorf("introduced position %d is outside the range of RDEs retrieved: [1, %d]", idx, len(rdes))
+		}
+		return rdes[idx-1], nil
+	}
+
+	log.Printf("[DEBUG] importing vcd_rde resource with provided id %s", d.Id())
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	var rde *govcd.DefinedEntity
+	var err error
+	switch len(resourceURI) {
+	case 1: // ie: urn:vcloud:entity:vendor:nss:a074f9e9-5d76-4f1e-8c37-f4e8b28e51ff
+		rde, err = vcdClient.VCDClient.GetRdeById(resourceURI[0])
+		if err != nil {
+			return nil, err
+		}
+	case 4: // ie: list@vendor.nss.1.2.3.name
+		listAndVendorSplit := strings.Split(resourceURI[0], "@")
+		if len(listAndVendorSplit) != 2 {
+			return nil, helpError
+		}
+		return nil, printList(listAndVendorSplit[1], resourceURI[1], resourceURI[2], resourceURI[3])
+	case 5: // ie: VCD_IMPORT_SEPARATOR="_" vendor_nss_1.2.3_name_1
+		rde, err = getRdeInPosition(resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3], resourceURI[4])
+		if err != nil {
+			return nil, err
+		}
+	case 6: // ie: VCD_IMPORT_SEPARATOR="_" list@vendor_nss_1.2.3_name
```

Comment:
It says "case 6", but the example has only 4 items.
It would be 6 items with the default import separator

Reply:
Good catch, it is indeed the case 4. I've fixed it.